src/editor/ui_menus/main_menu_map_options.cc

```diff
@@ -157,18 +305,31 @@ MainMenuMapOptions::MainMenuMapOptions(EditorInteractive& parent, Registry& regi
 	tags_box_.add(ww_box, UI::Box::Resizing::kFullSize);
 	tags_box_.add_space(padding_);
 
-	teams_box_.add(new UI::Textarea(&teams_box_, 0, 0, max_w_, labelh_, _("Suggested Teams:")));
-	teams_box_.add(&teams_list_);
-	// TODO(GunChleoc): We need team images in the listselect here,
-	// so map editors will be able to delete entries.
-	// This is waiting for the new RT renderer.
-	// TODO(Notabilis): Add onChanged-code below after this is added
-	teams_list_.add("Not implemented yet.", "", nullptr, false);
+	inner_teams_box_.set_force_scrolling(true);
+	inner_teams_box_.set_scrollbar_style(UI::PanelStyle::kWui);
+	for (const Widelands::SuggestedTeamLineup& team : parent.egbase().map().get_suggested_teams()) {
+		SuggestedTeamsEntry* ste = new SuggestedTeamsEntry(
+		   this, &inner_teams_box_, parent.egbase().map(), max_w_ - UI::Scrollbar::kSize, team);
+		inner_teams_box_.add(ste);
+		inner_teams_box_.add_space(kSuggestedTeamsUnitSize);
+		suggested_teams_entries_.push_back(ste);
+	}
 
+	teams_box_.add(new UI::Textarea(&teams_box_, 0, 0, max_w_, labelh_, _("Suggested Teams:")));
+	teams_box_.add(&inner_teams_box_, UI::Box::Resizing::kExpandBoth);
```

Comment:
Both the `teams_box_`and the `inner_teams_box_` are vertical, so you can scrap the inner one and just use the outer one.

Reply:
The layout is like this:
```
[begin teams_box_]
  "x Players"
  "Suggested teams:"
  [begin inner_teams_box_]
    team1
    team2
    …
  [end inner_teams_box_]
  "New entry"
[end teams_box_]
```

Since boxes append every new element to the end, we need to have an outer and an inner box here.

Reply:
I didn't see the append. All fine then :)